pkg/core/blockchain.go

```diff
@@ -1622,7 +1625,7 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 				stop = start + 1
 			}
 			for index := start; index < stop; index++ {
-				err := kvcache.DeleteBlock(bc.GetHeaderHash(index))
+				err := kvcache.DeleteBlock(bc.GetHeaderHash(index), bc.config.Ledger.RemoveUntraceableHeaders)
```

Comment:
Also, are you sure that we don't need some kind of special error for affected RPC handlers like `getblock`, `getrawtransaction`, `getstateroot`? It may be unobvious for the user who is unknown with this NeoGo extension why there's no data related to old headers.

Reply:
It's been there for a long long time already, if the node doesn't have data --- it returns "I have no data".